src/engine/universal/core/src/messaging-setup.js

```diff
@@ -58,4 +59,68 @@ module.exports = {
       }
     }
   },
+  async setupMonitoringAndLogging(messaging, configModule, machineModule, logger, network) {
+    let { serverAddress, baseTopic } = await configModule.readConfig('messaging');
+    if (!serverAddress) return;
+
+    if (baseTopic && !baseTopic.endsWith('/')) baseTopic += '/';
+    baseTopic += 'proceed-pms';
+
+    // Monitoring data
+    const [{ id: machineId }, loadInterval] = await Promise.all([
+      machineModule.getMachineInformation(['id']),
+      configModule.readConfig('engine.loadInterval'),
+    ]);
+
+    setInterval(async () => {
+      try {
+        const monitoring = await network.loopback('get', '/machine/');
```

Comment:
What exactly is t he reason for the `loopback` implementation? Couldn't this just be a call to the information submodule in `@proceed/machine`?

Reply:
Yeah, you make a good point. Initially I wanted to use the loopback for logging as well, because the endpoint was a bit complex, so I didn't want to reimplement it, but I ended up doing something else and kept using it for monitoring. For monitoring it doesn't make much sense, I'm going to change it, should I also remove the loopback?

Reply:
If it is not used in the end i think removing it would be the best.